File: API/Models/Message.cs

namespace API.Models
{
    public class Message
    {
        public int id { get; set; }
        public int senderid { get; set; }
        public User sender { get; set; }
        public int recipientid { get; set; }
        public User recipient { get; set; }
        public string content { get; set; }
        public bool is_read { get; set; }
        public DateTime? date_read { get; set; }
        public DateTime message_sent { get; set; } = DateTime.Now;
        public bool sender_deleted { get; set; }
        public bool recipient_deleted { get; set; }

    }
}

File: SPA/src/app/_core/_models/message.ts
export interface Message {
    id: number;
    senderid: number;
    sender_known_as: string;
    sender_photo_url: string;
    recipientid: number;
    recipient_known_as: string;
    recipient_photo_url: string;
    content: string;
    is_read: boolean;
    date_read: string | null;
    message_sent: string;
}